quinn/src/endpoint.rs

```diff
@@ -582,40 +540,32 @@ impl State {
     }
 }
 
-#[derive(Debug, Default)]
-struct TransmitState {
-    outgoing: VecDeque<udp::Transmit>,
-    /// The aggregateed contents length of the packets in the transmit queue
-    contents_len: usize,
-}
-
-impl TransmitState {
-    fn respond(&mut self, transmit: proto::Transmit, mut response_buffer: BytesMut) {
-        // Limiting the memory usage for items queued in the outgoing queue from endpoint
-        // generated packets. Otherwise, we may see a build-up of the queue under test with
-        // flood of initial packets against the endpoint. The sender with the sender-limiter
-        // may not keep up the pace of these packets queued into the queue.
-        if self.contents_len >= MAX_TRANSMIT_QUEUE_CONTENTS_LEN {
-            return;
-        }
-
-        let contents_len = transmit.size;
-        self.outgoing.push_back(udp_transmit(
-            transmit,
-            response_buffer.split_to(contents_len).freeze(),
-        ));
-        self.contents_len = self.contents_len.saturating_add(contents_len);
-    }
-
-    fn dequeue(&mut self, sent: usize) {
-        self.contents_len = self
-            .contents_len
-            .saturating_sub(self.outgoing.drain(..sent).map(|t| t.contents.len()).sum());
-    }
-
-    fn transmits(&self) -> &[udp::Transmit] {
-        self.outgoing.as_slices().0
-    }
+fn respond(socket: &dyn AsyncUdpSocket, transmit: proto::Transmit, response_buffer: &mut BytesMut) {
```

Comment:
I'm having trouble relating these changes to the commit message?

Reply:
This commit makes transmits stateless, hence `TransmitState` is no longer needed. Instead of buffering up outgoing endpoint-generated datagrams, we attempt to send them immediately and discard them if that fails.